sdk/core/azure-core-test/src/main/java/com/azure/core/test/TestBase.java

```diff
@@ -277,9 +277,9 @@ public static Stream<HttpClient> getHttpClients() {
      * <li>Otherwise, the name of the HttpClient class should match env variable.</li>
      * </ul>
      *
-     * Environment values currently supported are: "ALL", "netty", "okhttp" which is case insensitive.
+     * Environment values currently supported are: "ALL", "jdk", "netty", "okhttp" which is case insensitive.
      * Use comma to separate http clients want to test.
-     * e.g. {@code set AZURE_TEST_HTTP_CLIENTS = NettyAsyncHttpClient, OkHttpAsyncHttpClient}
+     * e.g. {@code set AZURE_TEST_HTTP_CLIENTS = JdkHttpClient, NettyAsyncHttpClient, OkHttpAsyncHttpClient}
```

Comment:
There needs to be further updates here to actually support JDK HttpClient. The original design isn't very flexible and requires a lot of updating to support different HttpClients.

Let's have a design overhaul here and have `AZURE_TEST_HTTP_CLIENTS` in the matrix pass the fully qualified `HttpProvider` class name and then leverage [HttpClientOptions.setHttpClientProvider](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/util/HttpClientOptions.java#L387) to configure the `HttpClient` used in testing. This way, if/when we add support for more `HttpClient`s in the future the only change required is adding it as a dependency in `azure-core-test` and everything will just work. For now, we'll still want to support `netty` and `okhttp` as options as removing this would require all SDKs to use from source `azure-core-test` which we don't want for now.

Additional note, not all Storage SDKs use `azure-core-test`'s logic here for determining which `HttpClient` to use, we'll need to work with them to update their test code while they transition over to Test Proxy and JUnit instead of local playback tests and Spock.